src/dialects/mssql/index.js

```diff
@@ -85,6 +85,15 @@ assign(Client_MSSQL.prototype, {
     })
   },
 
+  prepBindings(bindings) {
+    return map(bindings, (value) => {
+    if ( typeof value === 'number' && value >= 2147483647 || value <= -2147483648){
```

Comment:
I forgot to mention but yes this is to prevent loss of integer precision. 
so in fact tedious considers js number type are int4 by default. bigints are expected to be strings. I think tedious could be a bit smarter about the way it handles numbers by default. But until they improve that or we work on a PR, we will have to fix this on the knex side I think.